apps/generator/test/parser.test.js

```diff
@@ -104,4 +104,63 @@ describe('Parser', () => {
       expect(parsedDocument.document.version()).toEqual('3.0.0');
     });
   });
+  
+  describe('getProperApiDocument', () => {
+    it('should convert to old API if apiVersion is undefined', async () => {
+      const asyncapiDocument = {
+        version: '2.0.0',
+        json: jest.fn().mockReturnValue({ version: '2.0.0' })
+      };
+      const templateConfig = {};
+      const properDocument = getProperApiDocument(asyncapiDocument, templateConfig);
+      expect(properDocument).toBeDefined();
+      expect(properDocument._json.version).toEqual('2.0.0');
+    });
+
+    it('should convert to specified API version', async () => {
+      const asyncapiDocument = await parse(dummyV2Document, {}, {templateConfig: {apiVersion: 'v2'}});
+      const templateConfig = { apiVersion: 'v2' };
+      const properDocument = getProperApiDocument(asyncapiDocument, templateConfig);
+
+      expect(properDocument).toBeDefined();
+      expect(properDocument.document._json.asyncapi).toEqual('2.3.0'); 
+      expect(properDocument.diagnostics).toBeDefined();
+      expect(properDocument.diagnostics.length).toBeGreaterThan(0);
```

Comment:
how does this help to validate that conversion worked?

shouldn't you check if `properDocument` contains functions that are from the old API?